sdk/identity/identity/CHANGELOG.md

```diff
@@ -1,15 +1,129 @@
+- For `ClientCertificateCredential` specifically, now the validity of the PEM certificate is evaluated on `getToken` and not on the constructor.
+- For the `ManagedIdentityCredential` specifically:
+  - It now supports Bridge to Kubernetes local development authentication and token exchange authentication.
+  - It now retries with exponential back-off when a request for a token fails with a 404 status code on environments with available IMDS endpoints.
+  - Azure Service Fabric support hasn't been added on the initial version 2 of Identity. It will be added shortly after. You can follow our development of this feature through this issue: https://github.com/Azure/azure-sdk-for-js/issues/12420
+- For `InteractiveBrowserCredential`:
+  - On Node.js, we've enabled [Proof Key for Code Exchange (PKCE)](https://datatracker.ietf.org/doc/html/rfc7636). This is a security feature that mitigates authentication code interception attacks.
+  - We've also added a new `LoginHint` property to `InteractiveBrowserCredentialOptions` which allows a user name to be pre-selected for interactive logins. Setting this option skips the account selection prompt and immediately attempts to log in with the specified account.
+- For the `AzureCliCredential`, we now allow specifying a `tenantId` in the parameters through the `AzureCliCredentialOptions`.
+- Updates on our provided errors:
+  - We're introducing a new error, named `AuthenticationRequiredError`. This error will show up when a credential fails to authenticate silently. Users can now catch this error to know when to call to the new `authenticate()` method if any of the interactive credentials (`DeviceCodeCredential`, and `InteractiveBrowserCredential`) were initialized with `disableAutomaticAuthentication` constructor option set to true.
+  - Errors and logged exceptions may now point to the new [troubleshooting guidelines](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/Troubleshooting.md).
+
+The following example shows how to use the `DeviceCodeCredential` with the new `@azure/identity-cache-persistence` plugin. We are setting the `disableAutomaticAuthentication` option on the `DeviceCodeCredential` to manually trigger the user interaction. Then, we're using the new `@azure/identity-cache-persistence` plugin to enable sharing the same persistence cache between credential instances, and finally we're sharing the resulting `authenticationRecord` from a first `authenticate()` call with a new instance of the same credential to skip further user interactions on the next `getToken()` call.
+
+```ts
```

Comment:
instead of inlining this rather large code sample, can we place it somewhere in the source tree and link to it? That way it will get updated as needed

Reply:
I’m now making this way shorter and linking to the sample docs.